### PR TITLE
Search based on supported platforms

### PR DESCRIPTION
**Pull request by tuahk**
This starts work on issue #980.

I thought that I make my work so far public, in a hope of some feedback and ideas on how it should look like.

Idea is to add some kind of search filter where you could choose which platforms you want to include in search result.

### CODE REVIEW

**Comment by nellshamrell** [comment]
Looking good so far!

**Comment by nellshamrell** [comment]
How I imagine this working:

On the main page a user sees when they visit supermarket.chef.io, they will continue to see the prominent search field.  Beneath that search field, they will a see a link to "Advanced Search".  

When they click this link, they will then be taken to a view where they can somehow indicate platforms they wish the cookbooks returned to support.  Maybe a list of check boxes?  Text fields?  Drop downs?

**Comment by tuahk** [comment]
Ok, that sounds good. I'll try it with check boxes and see how it looks. There might be too many different supported platforms though.

**Comment by tuahk** [comment]
That latest commit adds a drop-down window where users can select which platforms to include in search result. 

Currently it should look something like this when closed:

screenshot - 26 04 2015 - 22 03 09

And like this when opened:

screenshot - 26 04 2015 - 22 09 10

There are still some issues with it but I start working with them if you guys like where this is going.

File: app/views/application/_search.html.erb [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
+                <label>
+                  <%=check_box_tag 'platforms[]', platform, params[:platforms] ? params[:platforms].include?(platform) : false %>
+                  <%= platform %>
+                </label>
+            </div>
+          <% end %>
+        </div>
+
+        <div class= "advanced_textfield">
+          <%= search_field_tag 'platforms[]', nil, placeholder: "You can write platform name here, if it's not on the list!", class: 'platform_search_textfield' %>
```
Comment (smith) [inline app/views/application/_search.html.erb]: I don't think the comma or the exclamation point are necessary here.

**Comment by smith** [comment]
@tuahk I like where this is going. Nice work so far!

**Comment by tuahk** [comment]
I removed WIP from this PR.

There are still functionalities that I feel should be added. For example advanced search should probably let users define multiple platforms in text input and keep what was written there after submit. But unfortunately I'm going to be super busy for the next couple of weeks and probably won't have time to implement those. 

However, I think this would already be a helpful feature for the supermarket and I'll of course try to find the time to fix any problems you might notice.

**Comment by nellshamrell** [comment]
Thanks so much!  Reviewing this now.

**Comment by nellshamrell** [comment]
Overall this looks great!  I worked on refining the styling a bit, do you mind incorporating these changes?  After that I think it's good to merge :)

https://github.com/tuahk/supermarket/compare/search-by-platforms...nellshamrell:search-by-platforms

**Comment by tuahk** [comment]
Great! Those changes made this look much better. Really liked that arrow :)

**Comment by nellshamrell** [comment]
Awesome.  Merging!

**Comment by nellshamrell** [comment]
This is now live!

https://supermarket.chef.io

**Pull request merged**